GitCommands/Settings/Setting.cs

```diff
@@ -8,14 +10,29 @@ public abstract class Setting<T>
         public readonly T DefaultValue;
         public readonly string Name;
 
+        public event Action Updated;
```

Comment:
It is customary to type events as `EventHandler` (or derivative)
```suggestion
        public event EventHandler SettingUpdated;
```